packages/slice-machine/src/legacy/lib/builders/common/Zone/Card/index.jsx

```diff
@@ -129,4 +139,28 @@ const FieldZone = ({
   );
 };
 
+function useScrollIntoLastAddedItem(fields) {
+  const isReadyRef = useRef(false);
+  const lastFieldsRef = useRef([]);
+  const lastItemRef = useRef(null);
+
+  useEffect(() => {
+    if (isReadyRef.current) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      if (fields != null && fields.length > lastFieldsRef.current.length) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        lastItemRef.current?.scrollIntoView({ behavior: "smooth" });
+      }
+    }
+  }, [fields]);
+
+  useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    lastFieldsRef.current = fields;
+    if (!isReadyRef.current) isReadyRef.current = true;
+  }, [fields]);
+
+  return { lastItemRef };
+}
+
```

Comment:
Not sure if this should be better implemented and typed under the features folder, WDYT @dani-mp? 👀

Reply:
Arf, this one is tricky, because a hook like that could be nice and simple in our hooks folder. Generic enough, so you pass an array of items and it scrolls to last one. WDYT? Is it really more complex?

Reply:
It's a bit dependent on the `fields` list, and we might be able to this in a more procedural way, instead of handling it as a side-effect no?

Reply:
Personally fine for me, it's today dependent to this specific design that will be changed, so I don't mind much.

Reply:
Got it, it's currently being changed anyway, so this hook doesn't exist anymore 😅